src/Microsoft.Azure.WebJobs.Host/Indexers/MethodInfoExtensions.cs

```diff
@@ -26,7 +28,15 @@ public static string GetShortName(this MethodInfo methodInfo)
                 throw new ArgumentNullException("methodInfo");
             }
 
-            return String.Format(CultureInfo.InvariantCulture, "{0}.{1}", methodInfo.DeclaringType.Name, methodInfo.Name);
+            return TryGetCustomName(methodInfo, out string customName)
+                ? customName
+                : String.Format(CultureInfo.InvariantCulture, "{0}.{1}", methodInfo.DeclaringType.Name, methodInfo.Name);
+        }
+
+        private static bool TryGetCustomName(MethodInfo methodInfo, out string customName)
```

Comment:
What guarantees are we making for this attribute as far as where the custom name takes effect? I see other places in the codebase that use function name w/o going through this helper, meaning they'd use the actual name, not attribute provided name.